v2/processor.go

```diff
@@ -169,3 +199,20 @@ func (plr *peekLockRenewer) startPeriodicRenewal(ctx context.Context, message *a
 		}
 	}
 }
+
+type MessageWrapper struct {
+	Message *azservicebus.ReceivedMessage
+}
+
+// Set implements tab.Carrier interface
+func (mw *MessageWrapper) Set(key string, value interface{}) {
+	if mw.Message.ApplicationProperties == nil {
```

Comment:
check if Message is nil too. we don't want this code to cause panics in user's message handling